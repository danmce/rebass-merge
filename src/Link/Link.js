import React from 'react'
import PropTypes from 'prop-types'
import { Link as RebassLink } from 'rebass'

const Link = ({ children, ...props }) => (
  <RebassLink {...props}>{children}</RebassLink>
)

Link.propTypes = {
  children: PropTypes.string,
  href: PropTypes.string,
  variant: PropTypes.oneOf(['nav'])
}

export default Link
