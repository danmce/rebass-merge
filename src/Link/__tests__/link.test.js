import React from 'react'
import * as enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import Link from '../Link'

test('renders Link correctly', () => {
  const link = <Link href='https://rebassjs.org'>Link</Link>

  expect(toJson(enzyme.mount(link))).toMatchSnapshot('enzyme.mount')
})
