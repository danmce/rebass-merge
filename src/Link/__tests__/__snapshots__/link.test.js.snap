// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Link correctly: enzyme.mount 1`] = `
<Link
  href="https://rebassjs.org"
>
  <ForwardRef
    href="https://rebassjs.org"
  >
    <Styled(div)
      as="a"
      href="https://rebassjs.org"
      variant="link"
    >
      <a
        className="emotion-0"
        href="https://rebassjs.org"
      >
        Link
      </a>
    </Styled(div)>
  </ForwardRef>
</Link>
`;
